scalafix-cli/src/main/scala/scalafix/internal/v1/Args.scala

```diff
@@ -242,9 +247,21 @@ case class Args(
     RuleDecoder.decoder(ruleDecoderSettings.withConfig(scalafixConfig))
   }
 
+  // With a --on-compile flag, looking for settings in onCompile block first, and fallback to standard settings.
+  def preProcessedConf(base: Conf): Conf =
+    if (onCompile) {
+      val confOnCompile = ConfGet.getKey(base, "onCompile" :: Nil)
+      confOnCompile.fold(base)(
+        ConfOps.merge(ScalafixConfOps.drop(base, "onCompile"), _)
+      )
```

Comment:
while you are at it, can't you extract the entire get/fold/merge logic to `ScalafixConfOps` so we better understand the intent here?